From d88efc58e283227a42c2f981304f33acb45bf204 Mon Sep 17 00:00:00 2001
From: ywan171 <yi.a.wang@intel.com>
Date: Sun, 5 May 2013 00:17:33 +0800
Subject: [PORT FROM MAIN]libmediaplayerservice: check surfaceTexture before nativewindow api disconnection or connection

BZ: 74797

nativewindow api disconnect or connect when surfaceTexture equals NULL will cause
TOMBSTONE error, should check surfaceTexture before native window api disconnection or
connection

Orig-Change-Id: I28c7cf9210a70d59a4226b282018332746f58722

Change-Id: I751e7dea4837cb82b0b807e13d66b6a2fea8cadb
Signed-off-by: ywan171 <yi.a.wang@intel.com>
---
 media/libmediaplayerservice/nuplayer/NuPlayer.cpp |   35 +++++++++++----------
 1 files changed, 18 insertions(+), 17 deletions(-)

diff --git a/media/libmediaplayerservice/nuplayer/NuPlayer.cpp b/media/libmediaplayerservice/nuplayer/NuPlayer.cpp
index f9e2092..00ce1db 100644
--- a/media/libmediaplayerservice/nuplayer/NuPlayer.cpp
+++ b/media/libmediaplayerservice/nuplayer/NuPlayer.cpp
@@ -225,24 +225,25 @@ void NuPlayer::setVideoSurfaceTextureAsync(
         msg->setObject("native-window", NULL);
     } else {
         sp<Surface> surface = new Surface(bufferProducer);
-        sp<ANativeWindow> anw = surface;
-        //NOTES: we must re-connect api here because we need to get right
-        //infomation from surface texture's back end. Otherwise, we only get
-        //uninitlized mTransformHint, mDefaultWidth, mDefaultHeight, etc.
-        status_t err = native_window_api_disconnect(anw.get(),
-                NATIVE_WINDOW_API_MEDIA);
-        if (err != OK) {
-            ALOGE("setVideoSurfaceTextureAsync: api disconnect failed: %d", err);
-            return;
-        }
+        if (surface != NULL) {
+            sp<ANativeWindow> anw = surface;
+            //NOTES: we must re-connect api here because we need to get right
+            //infomation from surface texture's back end. Otherwise, we only get
+            //uninitlized mTransformHint, mDefaultWidth, mDefaultHeight, etc.
+            status_t err = native_window_api_disconnect(anw.get(),
+                    NATIVE_WINDOW_API_MEDIA);
+            if (err != OK) {
+                ALOGE("setVideoSurfaceTextureAsync: api disconnect failed: %d", err);
+                return;
+            }
 
-        err = native_window_api_connect(anw.get(),
-                NATIVE_WINDOW_API_MEDIA);
-        if (err != OK) {
-            ALOGE("setVideoSurfaceTextureAsync: api connect failed: %d", err);
-            return;
+            err = native_window_api_connect(anw.get(),
+                    NATIVE_WINDOW_API_MEDIA);
+            if (err != OK) {
+                ALOGE("setVideoSurfaceTextureAsync: api connect failed: %d", err);
+                return;
+            }
         }
-
         msg->setObject(
                 "native-window",
                 new NativeWindowWrapper(surface));
-- 
1.7.4.1

